isis/src/base/apps/cam2cam/cam2cam.h

```diff
@@ -41,6 +53,132 @@ namespace Isis {
       int OutputSamples() const;
       int OutputLines() const;
   };
+
+
+  /**
+   * @brief Standalone cam2cam pixel mapper functor
+   * 
+   * This class provides an instance of a standalone implementation of the
+   * cam2cam mapping algorithm. It is designed to provide single pixel mapping
+   * of the match/to output image to the from image. Note that the MATCH and TO
+   * cubes have exactly the same geometry so TO is not required for this 
+   * implementation.
+   * 
+   * The cam2cam algorithm is output driven by mapping every sample/line pixel
+   * coordinate in the match image back into the from image FOV.
+   * 
+   * This class functor is designed to just compute the coordinate in the FROM
+   * image and determine how it is mapped into the FROM/MATCH paramterization.
+   * This is well suited for testing the algorithm without significant overhead
+   * of processing, typically large, ISIS cubes.
+   * 
+   * It does not create an output cube or interpolation algorithms. Selection of
+   * input bands is entirely managed by the owner of this object.
+   * 
+   * Note this should *not* be used as the processing algorithm directly in
+   * cam2cam because it does not use the RubberSheet mapping algorithm as the
+   * main app uses. This essentially uses the RubberSheet::SlowGeom() method
+   * (w/out interpolation!), which is the least efficicent algorithm.
+   * However it may be applicable in a threaded implementation.
+   * 
+   * @author 2024-10-27 Kris J. Becker
+   */
+  class Cam2CamMapper {
```

Comment:
Since this class is only used in the tests, can it be moved into ```FunctionalTestsCam2cam.cpp```?

Reply:
I think this class has general utility other than strictly testing. All one has to do is add the include file in their own code.

It also keeps the class with the context in which it is designed to model.

Reply:
If cam2cam was updated to use it I would agree but it isn't currently being used in the codebase, only the test suite.

It also seems like a rewrite of the cam2camXform, could one not use the cam2camXform in the same way that one would use this new class?

Reply:
Yeah, you may be right.

It is odd that it appears the only way to use this externally would require a link to cam2cam.o.  It would be more better/useful if the implementation of cam2camXform is entirely in the header.

Reply:
We could move the xform implementation into the .h as well